src/utils/actions/writeAction.ts

```diff
@@ -1,24 +1,65 @@
-import { mutate } from '~amplifyClient';
-import { ContractEvent } from '~types';
+import { Extension, getExtensionHash } from '@colony/colony-js';
+import { mutate, query } from '~amplifyClient';
+import { ColonyActionType, ContractEvent } from '~types';
 import { verbose } from '~utils';
 
+interface ActionFields extends Record<string, any> {
+  initiatorAddress: string;
+  type: ColonyActionType;
+}
+
 export const writeActionFromEvent = async (
   event: ContractEvent,
   colonyAddress: string,
-  actionFields: Record<string, any>,
+  actionFields: ActionFields,
 ): Promise<void> => {
   const { transactionHash, blockNumber, timestamp } = event;
 
   const actionType = actionFields.type ?? 'UNKNOWN';
-  verbose('Action', actionType, 'took place in Colony:', colonyAddress);
+  const showInActionsList = await showActionInActionsList(
+    colonyAddress,
+    actionFields.initiatorAddress,
+  );
 
+  verbose('Action', actionType, 'took place in Colony:', colonyAddress);
   await mutate('createColonyAction', {
     input: {
       id: transactionHash,
       colonyId: colonyAddress,
       blockNumber,
       createdAt: new Date(timestamp * 1000).toISOString(),
+      showInActionsList,
       ...actionFields,
     },
   });
 };
+
+const getVotingReputationInstallations = async (
+  colonyAddress: string,
+): Promise<Array<{ id: string }> | undefined> => {
+  const votingRepHash = getExtensionHash(Extension.VotingReputation);
+  const { items } =
+    (await query<{ items: Array<{ id: string }> }>(
+      'getVotingRepInstallations',
+      {
+        votingRepHash,
+        colonyAddress,
+      },
+    )) ?? {};
+
+  return items;
+};
+
+const showActionInActionsList = async (
+  colonyAddress: string,
+  initiatorAddress: string,
+): Promise<boolean> => {
+  const votingRepIds = await getVotingReputationInstallations(colonyAddress);
```

Comment:
I wanted to complain about this, and have you fetch the extension from the chain directly (since you can just get the address directly, and it'll be faster than graphql) but I've come to understand your approach is better.

If you use my approach, you can only filter using the currently installed extension address (if it exists), all other actions, for earlier uninstalled extensions would show up at that point.

So your approach is better in this case :)